pkg/kubelet/kubelet_node_status.go

```diff
@@ -599,9 +599,7 @@ func validateNodeIP(nodeIP net.IP) error {
 	for _, addr := range addrs {
 		var ip net.IP
 		switch v := addr.(type) {
-		case *net.IPNet:
-			ip = v.IP
-		case *net.IPAddr:
+		case *net.IPNet, *net.IPAddr:
```

Comment:
Could you share more about the advantages of a `case` statement vs a single `if` statement, given that we only have a single block in our case statement?

Reply:
My change here is to merge two cases with the same treatment into one. As for the comparison with if, I think it is more concise than the code implemented by if.

Reply:
Ok - curious if other have strong opinions. At the end of the day, I think both are better than a switch statement with two duplicate cases.

Do you have thoughts on the error message we are getting when running `govet`? 

```
pkg/kubelet/kubelet_node_status.go:603:10: v.IP undefined (type "net".Addr has no field or method IP)
```

From https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/90100/pull-kubernetes-verify/1249579974818533377.

Reply:
I did not notice this problem before, it seems that this problem will not occur before the change. This is very strange, and I don’t quite understand the cause of this problem.

Reply:
After understanding the go `type switch`, I think this modification is wrong, the original code does not need to be modified.